### PR TITLE
doc: Add ABP vs OTAA doc

### LINKED ISSUE #215: Document why OTAA is better than ABP
#### Summary

We need to write some documentation that recommends OTAA over ABP.

#### Why do we need this ?

Until last week, we didn't, because most of our Cloud customers already use OTAA. However, we're now seeing a large number of TTN users starting to migrate devices to v3, and many of those are ABP.

#### What is already there? What do you see now?

There's a lot of documentation (in the wild) that uses ABP. This is mostly because it used to be easier (simpler) to start with ABP. There are also a lot of ABP devices on TTN v2 that will (hopefully not) get migrated to TTN v3.

#### What is missing? What do you want to see?

We need to document the limitations of ABP and explain why it's better to use OTAA.

#### How do you propose to document this?

- ABP devices use a fixed DevAddr
	- This means you can not switch to a different network or cluster.
		- Even if a network or cluster allows you to register end devices with an DevAddr that isn't assigned to the cluster, Packet Broker won't route traffic from that end device to the right cluster.
	- This means the Network Server can't optimize DevAddr allocation when additional blocks are assigned to a cluster. As a result, device matching will not improve for these devices.
- ABP devices use a fixed security session
	- This means that a device is EOL when it runs out of frame counters
	- This means that session keys can not be rotated if leaked
- ABP devices use fixed network parameters
	- This means that the RX1 delay, RX1 DR offset, RX2 DR index, RX2 Frequency and channel frequencies need to be customized, and registered on the Network Server. If not configured exactly right, uplinks/downlinks may not work, and this will be very difficult to debug.
- ... 

Let's discuss what else we can say

### PR DESCRIPTION
**Pull request by nejraselimovic**
#### Summary
Adding a section dedicated to accenting why OTAA is recommended and not ABP.
Closes #215 

#### Notes for Reviewers
More info from the team is warmly welcome

#### Checklist
- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+This section can help you understand the differences between ABP and OTAA activation modes for LORaWAN end devices, and comprehend why using OTAA is recommended. 
+
+<!--more-->
+
+A `DevEUI` is a 64-bit unique ID assigned to an end device by the manufacturer. This value is linked to the hardware and it cannot be altered.
+
+Unlike `DevEUI`, which identifies an end device globally, a 32-bit `DevAddr` identifies the end device within the current network and all communication after joining the network is done with it. A `DevAddr` value consists of `NwkID` and `NwkAddr`. `NwkID` is a 7-bit network identifier, used to distinguish (overlapping) LoRaWAN networks, while `NwkAddr` is a 25-bit network address of the end device.
+
+{{< note >}} `DevAddr` are not unique identifiers. If there are more than one device with identical `DevAddr`, {{% tts %}} Network Server has to perform **matching**. See more in the [Network Server reference]({{< ref "/reference/components/network-server#uplink-handling" >}}) section. {{</ note >}}
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:14]: I think this is a bit confusing. The Network Server always performs **matching**, even if there is only one device with that DevAddr.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+description: ""
+---
+
+This section can help you understand the differences between ABP and OTAA activation modes for LORaWAN end devices, and comprehend why using OTAA is recommended. 
+
+<!--more-->
+
+A `DevEUI` is a 64-bit unique ID assigned to an end device by the manufacturer. This value is linked to the hardware and it cannot be altered.
+
+Unlike `DevEUI`, which identifies an end device globally, a 32-bit `DevAddr` identifies the end device within the current network and all communication after joining the network is done with it. A `DevAddr` value consists of `NwkID` and `NwkAddr`. `NwkID` is a 7-bit network identifier, used to distinguish (overlapping) LoRaWAN networks, while `NwkAddr` is a 25-bit network address of the end device.
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:12]: I think this 7+25 is only true for type 0 NetIDs.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+{{< note >}} `DevAddr` are not unique identifiers. If there are more than one device with identical `DevAddr`, {{% tts %}} Network Server has to perform **matching**. See more in the [Network Server reference]({{< ref "/reference/components/network-server#uplink-handling" >}}) section. {{</ note >}}
+
+A `DevAddr` and security keys are assigned to an end device during a procedure called **activation**. LoRaWAN supports two modes of activating an end device:
+
+- ABP (Activation By Personalization)
+- OTAA (Over-The-Air Activation)
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:19]: I don't think we like such bullet lists, maybe just pull them into the paragraph above.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+ABP's drawbacks are arising as consequences of its main characteristics.
+
+1. **ABP end devices use a fixed `DevAddr`.**
+
+  Taking the structure of `DevAddr` into the account and the fact that, with ABP, `NwkID` is fixed, we come to the conclusion that this device can work correctly only in its predefined network. Even if a network or a cluster allows registering end devices with different `DevAddr` values than the ones that were assigned to that network/cluster, the [Packet Broker]({{< ref "/reference/peering#packet-broker" >}}) will not route the traffic from those end devices to the right network/cluster. 
+
+  Also, even if the network/cluster is assigned with additional address blocks, the Network Server will not be able to perform optimization of `DevAddr` allocation, hence the device matching procedure for ABP end devices will not be improved.
+
+  OTAA end devices are assigned a new `DevAddr` at establishing each new session.
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:39]: Maybe add that this also allows OTAA devices to move to a different clusters or networks.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+2. **ABP end devices use a fixed security session.**
+
+  ABP end devices must not reset frame counters during their lifetime, otherwise the messages from those end devices may end up being dropped. Not being able to reset frame counters means that the end of life for the ABP end device is when it runs out of frame counters.
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:43]: It sounds a bit like frame counters are the bad guys here.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  Session keys are hardcoded into an ABP device, meaning that even if they leak, they cannot be rotated, which is a serious security threat.
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:45]: Maybe a good place to link to some information about how Join Servers and Secure Elements make everything better.

File: doc/content/devices/abp-vs-otaa/_index.md (hunk 1 above)
Comment (johanstokking) [reply doc/content/devices/abp-vs-otaa/_index.md:14]: Let's keep this doc simple also; that NS reference is tough stuff, while this page should be really entry level

File: doc/content/devices/abp-vs-otaa/_index.md (hunk 2 above)
Comment (johanstokking) [reply doc/content/devices/abp-vs-otaa/_index.md:12]: Indeed, just say here that DevAddr are prefixed by a network identifier for identification.

File: doc/content/devices/abp-vs-otaa/_index.md (hunk 6 above)
Comment (nejraselimovic) [reply doc/content/devices/abp-vs-otaa/_index.md]: I linked some info, but there isn't much on our documentation site at the moment.. Are these being used with both type of devices?

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+description: ""
+---
+
+This section can help you understand the differences between ABP and OTAA activation modes for LORaWAN end devices, and comprehend why using OTAA is recommended. 
+
+<!--more-->
+
+A `DevEUI` is a 64-bit unique ID assigned to an end device by the manufacturer. This value is linked to the hardware and it cannot be altered.
+
+Unlike `DevEUI`, which identifies an end device globally, a 32-bit `DevAddr` identifies the end device within the current network and all communication after joining the network is done with it. A `DevAddr` value consists of `NwkAddr` (end device address within network) prefixed by a `NwkID` (network identifier).
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:12]: I think we should mention that multiple (many even) end devices can have the same `DevAddr`.

File: doc/content/devices/abp-vs-otaa/_index.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+A `DevAddr` and security keys are assigned to an end device during a procedure called **activation**. LoRaWAN supports two modes of activating an end device: **ABP (Activation By Personalization)** and **OTAA (Over-The-Air Activation)**.
```
Comment (htdvisser) [inline doc/content/devices/abp-vs-otaa/_index.md:14]: Maybe use "session keys" here as well.

**Review by htdvisser — APPROVED** [review]
It may not be perfect, but it's already a lot better than not having anything, so let's get this in and make more improvements later.

**Review by johanstokking — APPROVED** [review]
Nothing to add, rest looks great

**Pull request merged**